File: src/net/fodt/Spec/Text/Heading.cs
// FormatFactory.Fodt — Spec.Text.Heading — Canonical spec-shaped model class
// spec_qname: text:h
// spec_fact_ref: FACT-FODT-004
// TC-QHARD-051: converted from architecture_only stub to real model class
namespace FormatFactory.Fodt.Spec.Text;

/// <summary>
/// Spec-shaped model class for the ODF text:h (heading) element.
///
/// ODF 1.3 §5.1.2 — text:h is the canonical heading element in ODF text
/// documents. The text:outline-level attribute specifies the heading depth (1–10).
/// spec_qname: text:h
/// spec_fact_ref: FACT-FODT-004
/// </summary>
public sealed class Heading
{
    /// <summary>The ODF QName for this element. Grounded in ODF 1.3 §5.1.2.</summary>
    public const string SpecQName = "text:h";

    /// <summary>The SAL fact reference for this element (sal-facts-latest.json).</summary>
    public const string SpecFactRef = "FACT-FODT-004";

    /// <summary>
    /// The text:outline-level attribute — heading depth, 1–10.
    /// Per ODF 1.3 §5.1.2.
    /// </summary>
    public int OutlineLevel { get; init; } = 1;

    /// <summary>The plain-text content of this heading element.</summary>
    public string Content { get; init; } = string.Empty;

    /// <summary>The text:style-name attribute value, if present. May be null.</summary>
    public string? StyleName { get; init; }
}

File: src/python/fodt/spec/text/heading.py
"""FODT spec Heading — canonical implementation of text:h.

spec_qname: text:h
spec_fact_ref: FACT-FODT-004
"""
from __future__ import annotations
from typing import Any, ClassVar

from .span import Span


class Heading:
    """Canonical implementation of ODF text:h element."""

    spec_qname: ClassVar[str] = "text:h"
    spec_fact_ref: ClassVar[str] = "FACT-FODT-004"

    def __init__(self, data: dict[str, Any]):
        self._data = data

    @property
    def kind(self) -> str:
        return "heading"

    @property
    def text(self) -> str:
        return self._data.get("text", "")

    @property
    def style_name(self) -> str:
        return self._data.get("style_name", "")

    @property
    def outline_level(self) -> int | None:
        return self._data.get("outline_level")

    @property
    def spans(self) -> list[Span]:
        return [Span(s) for s in self._data.get("spans", [])]

    def to_dict(self) -> dict[str, Any]:
        return dict(self._data)

    def __repr__(self) -> str:
        return f"Heading(level={self.outline_level!r}, text={self.text!r})"
